### DISCUSSION TITLE
Docker Image

### DISCUSSION THREAD

**Discussion by itsddpanda** [Gkiokan/ps4-remote-pkg-sender#61] [General]
Hey All,

Is there an docker image available for this ?

If yes, can you share where I can find it.

TIA.

**Comment by Gkiokan** [comment]
Hi Panda!  

Currently there is no Docker Image for this application.  
But it is bundled as single executable binary, there are no install instructions at all.  
Download, Allow, Start, Have fun.  
  
May I ask whats the Idea behind using the application in a Docker Image?  
This would take way more recourses then starting the application itself.

**Reply by itsddpanda** [reply]
Will be happy to share details. 

So in my home network I got a NAS, I already have a home server running as VM on the same. And many docker images use NAS mounts for storage and it is pretty good so far. 

Now this VM is not a windows machine and I have no idea how (or if) the source code can be run on linux distro, i am Ubuntu btw. 

I already have docker and thought if this application could be run in it and has web-gui like all the other applications i do not have to keep my personal laptop hooked to LAN (to server PS4 files from NAS). Wifi is not very reliable and slow too.

That way i will have this application running in docker serving the PS4 files to the 3 PS4 already connected to LAN.

Come to think of it, this is pretty useful use case of (docker implementation) this application..

**Reply by Gkiokan** [reply]
The application does run on Linux, too. Just pick your favorite distribution package.  
It should be possible to put your Share Drive as the base Path and it would provide the files from NAS to your Installer on the ps4. 

But for your usecase my [HB-Store-CDN-CLI-Server](https://github.com/Gkiokan/hb-store-cdn-cli-server) would be a better fit. 
It is even supported by retroNAS, so you can hookup all other ROM's, too - just in case you are using retroNAS.

**Reply by GitHubber1312** [reply]
I have a somewhat similar problem. I would also like to install files from my NAS (unraid) onto the PS4. So far, I’ve always started the Windows exe on my desktop and then transferred the data via SMB shares to the RPSv2. But I would prefer to do everything without my desktop and thought about setting up a Docker container that I could then access via the browser (e.g., on my smartphone) to install games. After 10 hours of trying and failing, I’m giving up.

The HB-Store option is good, but the HB-Store app is (at least for me) quite slow when it comes to the interface. That’s why I wanted to use the nice, fast PS4 Remote PKG Sender. So if anyone manages to build a Docker container or has tips for me, I’d be very grateful!

or there is this gamebato thing... could we use that gui vor our pkg´s? any idea?

Still, I want to say thank you for the great work!

**Reply by Gkiokan** [reply]
The install Speed in HB-Store shouldn't be way more different then using my tool and RPI on the ps4 because everything is using the same technical function to install the games. Using the HB-Store or even ItemzFlow would help in your case. Just spin up my HB Store CLI Server on your NAS and you can do the rest on your ps4 via HB-Store - I have even RetroNAS Support!

There will be somewhen a CLI Client Version of my app, with GoldHEN Support for better and stable Installations, but no ETA yet. We are all busy here and there.

**Reply by GitHubber1312** [reply]
No, no, the installation is just as fast. I have 415 files (since there’s no distinction between updates, DLC, and so on) showing in the list, and I think the HB Store app can’t load that many images. From page 13 onward, the app crashes.

**Reply by Gkiokan** [reply]
HB-Store crashes? Can you check the logs and share us on pkg-zone Discord server? We need to investigate why it crashes. Maybe there is an issue with my tool or the hb-store has another strange bug on such scale.

**Reply by pengytek** [reply]
> I have a somewhat similar problem. I would also like to install files from my NAS (unraid) onto the PS4. So far, I’ve always started the Windows exe on my desktop and then transferred the data via SMB shares to the RPSv2. But I would prefer to do everything without my desktop and thought about setting up a Docker container that I could then access via the browser (e.g., on my smartphone) to install games. After 10 hours of trying and failing, I’m giving up.
> 
> The HB-Store option is good, but the HB-Store app is (at least for me) quite slow when it comes to the interface. That’s why I wanted to use the nice, fast PS4 Remote PKG Sender. So if anyone manages to build a Docker container or has tips for me, I’d be very grateful!
> 
> or there is this gamebato thing... could we use that gui vor our pkg´s? any idea?
> 
> Still, I want to say thank you for the great work!

Did you ever figure out something for Unraid?

**Reply by Gkiokan** [reply]
I think it works when you bind it locally as Network storage and provide that path as base Path in the app. That should work. I didn't tried myself as I don't have a NAS myself, but technically it should work. Others did the same with google drives.

**Reply by minkcodin** [reply]
I have the same idea of this app having a Docker Image.
The main idea is that I usually download games via a home server, which is running 24/7. But the config in the app only allows me to pick the IP of the PC, but the server is keeping all the pkg files, and I want to install it from the server, not from the PC
